src/main/java/org/jitsi/videobridge/RtpChannel.java

```diff
@@ -1857,13 +1857,38 @@ public void setSources(List<SourcePacketExtension> sources)
         addedSSRCs.removeAll(oldSignaledSSRCs);
         if (!addedSSRCs.isEmpty())
         {
+
+            boolean addToSyncronizer = false;
+            if (getContent().isRecording())
+            {
+                Recorder recorder = getContent().getRecorder();
+                Endpoint endpoint = getEndpoint();
+                if (recorder != null && endpoint != null)
+                {
+                    Synchronizer synchronizer = recorder.getSynchronizer();
+                    if (synchronizer != null)
+                    {
+                        addToSyncronizer = true;
+                    }
+                }
+            }
+
             for (Integer addedSSRC : addedSSRCs)
             {
                 try
                 {
                     // Do allow the number of explicitly signalled SSRCs to
                     // exceed the limit.
                     addReceiveSSRC(addedSSRC, false);
+
+                    // If recording is enabled, we need to add the mapping from
+                    // SSRC to EndpointID into the Synchronizer
+                    // instance used by RecorderRtpImpl.
+                    if (addToSyncronizer)
+                    {
+                        synchronizer.setEndpoint(addedSSRC & 0xffffffffl,
```

Comment:
The `synchronizer` variable will be out-of-scope before you reach this code block.